**First Inspection**

In [174]:
import pandas as pd
import numpy as np

pd.set_option("display.max_columns", None)
pd.set_option("display.max_rows", 50)

In [175]:
breweries = pd.read_csv("../raw data/breweries.csv")

In [176]:
breweries.head()

id                     name brewery_type  \
0  5128df48-79fc-4f0f-8b52-d06be54d0cec         (405) Brewing Co        micro   
1  9c5a66c8-cc13-416f-a5d9-0a769c87d318         (512) Brewing Co        micro   
2  34e8c68b-6146-453f-a4b9-1f6cd99a5ada  1 of Us Brewing Company        micro   
3  6d14b220-8926-4521-8d19-b98a2d6ec3db     10 Barrel Brewing Co        large   
4  e2e78bd8-80ff-4a61-a65c-3bfbd9d76ce2     10 Barrel Brewing Co        large   

                     address_1 address_2 address_3            city  \
0               1716 Topeka St       NaN       NaN          Norman   
1        407 Radam Ln Ste F200       NaN       NaN          Austin   
2          8100 Washington Ave       NaN       NaN  Mount Pleasant   
3                62970 18th St       NaN       NaN            Bend   
4  1135 NW Galveston Ave Ste B       NaN       NaN            Bend   

  state_province postal_code        country       phone  \
0       Oklahoma  73069-8224  United States  4058160490   
1          Texas  78745-1197  United States  5129211545   
2      Wisconsin  53406-3920  United States  2624847553   
3         Oregon  97701-9847  United States  5415851007   
4         Oregon  97703-2465  United States  5415851007   

                    website_url   longitude   latitude  
0     http://www.405brewing.com  -97.468182  35.257389  
1     http://www.512brewing.com         NaN        NaN  
2  https://www.1ofusbrewing.com  -87.883364  42.720108  
3       http://www.10barrel.com -121.281706  44.086835  
4                           NaN -121.328802  44.057565

In [177]:
breweries.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 9083 entries, 0 to 9082
Data columns (total 14 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   id              9083 non-null   object 
 1   name            9083 non-null   object 
 2   brewery_type    9083 non-null   object 
 3   address_1       8327 non-null   object 
 4   address_2       277 non-null    object 
 5   address_3       39 non-null     object 
 6   city            9083 non-null   object 
 7   state_province  9083 non-null   object 
 8   postal_code     9083 non-null   object 
 9   country         9083 non-null   object 
 10  phone           8156 non-null   object 
 11  website_url     7856 non-null   object 
 12  longitude       6735 non-null   float64
 13  latitude        6735 non-null   float64
dtypes: float64(2), object(12)
memory usage: 993.6+ KB


**Create a working copy (never work on the actual raw dataset)**

In [178]:
df = breweries.copy()

**Standardise column names**

In [179]:
df.columns = (
    df.columns
    .str.lower()
    .str.strip()
)

**Drop columns you don’t need**

In [180]:
df = df[
    [
        "id",
        "name",
        "brewery_type",
        "city",
        "state_province",
        "country",
        "latitude",
        "longitude",
    ]
]

**Quick check**

In [181]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 9083 entries, 0 to 9082
Data columns (total 8 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   id              9083 non-null   object 
 1   name            9083 non-null   object 
 2   brewery_type    9083 non-null   object 
 3   city            9083 non-null   object 
 4   state_province  9083 non-null   object 
 5   country         9083 non-null   object 
 6   latitude        6735 non-null   float64
 7   longitude       6735 non-null   float64
dtypes: float64(2), object(6)
memory usage: 567.8+ KB


**Sanity check the country column**

#How US-heavy is the dataset? 

In [182]:
df["country"].value_counts().head(15)

country
United States    8126
Australia         514
South Africa      104
Ireland            70
England            62
South Korea        61
Poland             34
Singapore          33
Austria            15
Portugal           14
Japan              10
Germany            10
Sweden             10
Scotland           10
Italy               4
Name: count, dtype: int64

## Data Coverage Observation

The dataset is heavily skewed toward the United States, which accounts for the vast majority
of recorded breweries. This reflects the underlying data collection process of Open Brewery DB,
which relies on open and community-sourced contributions. As a result, brewery counts should
be interpreted as indicators of dataset coverage rather than true national totals.

## Analytical Scope: Excluding the United States

Due to the overwhelming dominance of U.S. entries in the dataset, the United States is excluded
from subsequent analyses. This allows for a more interpretable comparison between other countries
represented in Open Brewery DB.

All results should be interpreted as reflecting dataset coverage rather than true national brewery counts.

In [183]:
df_non_us = df[df["country"] != "United States"].copy()

**Quick Check**

In [184]:
df_non_us.shape

(957, 8)

In [185]:
df_non_us["country"].value_counts()

country
Australia       514
South Africa    104
Ireland          70
England          62
South Korea      61
Poland           34
Singapore        33
Austria          15
Portugal         14
Japan            10
Germany          10
Sweden           10
Scotland         10
Italy             4
France            3
Isle of Man       2
Ukraine           1
Name: count, dtype: int64

## First analysis

# Top tier (within this dataset)

- Australia (514) → very strong coverage, active craft scene, English-language bias

- South Africa (104) → surprisingly high → interesting insight! 

- Ireland (70) → culturally expected, validates the dataset a bit

# UK fragmentation

- England (62)

- Scotland (10)

- Isle of Man (2)

UK ≈ 74 breweries -> That would put the UK ahead of Ireland in this dataset.

## Country Definitions and Cultural Context

The dataset distinguishes between England, Scotland, and other constituent countries
of the United Kingdom. These categories are preserved in the analysis.

This decision is intentional: brewing traditions, alcohol culture, and identity
differ significantly across these regions, and aggregating them into a single
"United Kingdom" category would obscure culturally meaningful differences.

In [186]:
# cleaned dataset including the US
df.to_csv(
    "../cleaned data/breweries_cleaned_all.csv",
    index=False
)

## Cleaning Complete

The cleaned dataset was saved to disk and will be used for all subsequent analyses
and visualisations. This notebook focuses solely on inspection and data preparation.